gcode_metadata/metadata.py

```diff
@@ -238,14 +239,46 @@ def cache_name(self):
         return new_path
 
     def is_cache_fresh(self):
-        """If cache is fresher than file, returns True"""
+        """Checks if we can use the current cache file"""
+        return self.is_cache_recent() and self.is_cache_correct_version()
+
+    def is_cache_recent(self):
+        """Checks if the cache file is newer than the source file"""
         try:
             file_time_created = os.path.getctime(self.path)
             cache_time_created = os.path.getctime(self.cache_name)
-            return file_time_created < cache_time_created
         except FileNotFoundError:
             return False
 
+        return file_time_created < cache_time_created
+
+    def is_cache_correct_version(self):
+        """Checks if the cache file was created with the same version
+        of gcode-metadata"""
+
+        def isallowed(char):
+            """Filters out disallowed characters, used with str.filter"""
+            return char not in "\",\n} "
+
+        # This expects the first item in the json file to be the
```

Comment:
kdyz ocekáváš že to má být v první line je potřeba v případě nenajití toho stringu procházet celý soubor?

Reply:
Načítám soubor řádek o řádku, jakmile narazím na jakékoli písmenko abecedy, zjistím, jestli načtené znaky začínají slovem "version", pokud ano, pokračuju parsováním verze. Pokud ne, vracím false. Jediný případ kdy bych přečetl celý soubor by nastal, pokud by v něm nebyla písmenka, což se v normálním případě nestane, jen někdo se zlým úmyslem by mi mohl soubor zaplnit pouze speciálnímy znaky a čísly a v takovém případě bych doopravdy četl celý soubor.
Pokud by se celý soubor nacházel na prvním řádku, přečtu ho taky celý, to není ideální, ale dokud s tím nebude problém, nechci špinit kód dalšími checky